igor-web/src/main/groovy/com/netflix/spinnaker/igor/wercker/WerckerBuildMonitor.groovy

```diff
@@ -0,0 +1,335 @@
+    }
+
+    @PreDestroy
+    void stop() {
+        log.info('Stopped')
+        if (!worker.isUnsubscribed()) {
+            worker.unsubscribe()
+        }
+    }
+
+    /**
+     * Gets a list of pipelines for this master & processes runs between last poll stamp and a sliding upper bound stamp,
+     * the cursor will be used to advanced to the upper bound when all builds are completed in the commit phase.
+     */
+    @Override
+    protected PipelinePollingDelta generateDelta(PollContext ctx) {
+        String master = ctx.partitionName
+        log.info("Checking for new builds for $master")
+        def startTime = System.currentTimeMillis()
+
+        List<PipelineDelta> delta = []
+
+        WerckerService werckerService = buildMasters.map[master] as WerckerService
+        long since = System.currentTimeMillis() - (Long.valueOf(getPollInterval() * 2 * 1000))
+        try {
+            if (front50.isPresent()) {
+                Set<String> pipelines = getConfiguredJobs(master)
+                Map<String, List<Run>> runs = werckerService.getRunsSince(pipelines, since)
+                runs.keySet().forEach( { pipeline ->
+                    processRuns(werckerService, master, pipeline, delta, runs.get(pipeline))
+                } )
+            } else {
+                Map<String, List<Run>> runs = werckerService.getRunsSince(since)
+                runs.keySet().forEach( { pipeline ->
+                    processRuns(werckerService, master, pipeline, delta, runs.get(pipeline))
+                } )
+            }
+        } catch (e) {
+            log.error("Error processing runs for Wercker[{}]", kv("master", master), e)
+        }
+        log.debug("Took ${System.currentTimeMillis() - startTime}ms to retrieve Wercker pipelines (master: {})", kv("master", master))
+        return new PipelinePollingDelta(master: master, items: delta)
+    }
+
+    Set<String> getConfiguredJobs(String master) {
+        Set<String> jobs = []
+        front50.get().getAllPipelineConfigs().forEach({ pipeline ->
+            pipeline['triggers'].forEach({ trigger ->
+                if (trigger['enabled'] && trigger['type'] == 'wercker' && trigger['master'] == master) {
+                    log.debug "configured Trigger for ${master} pipeline: ${pipeline['application']} ${pipeline['name']} ${trigger}"
+                    jobs.add(trigger['job'])
+                }
+            })
+            pipeline['stages'].forEach({ stage ->
+                if (stage['type'] == 'wercker' && stage['master'] == master) {
+                    log.debug "configured Stage for ${master} pipeline: ${pipeline['application']} ${pipeline['name']} ${stage}"
+                    jobs.add(stage['job'])
+                }
+            })
+        })
+        return jobs
+    }
+
+    static Comparator<Run> finishedAtComparator = new Comparator<Run>() {
+        @Override
+        public int compare(Run r1, Run r2) {
```

Comment:
The `finishedAtComparator` and `startedAtComparator` are almost duplicated---it might be easier to use something in the `Comparator` library rather than implement this manually.

Something like `Comparator.nullsLast({ r -> r.finishedAt?.fastTime })` should be equivalent and would make the intent more clear.

Reply:
Fixed. The comparators are now constants (of Run) to be reused.